utam-compiler/src/test/java/utam/compiler/grammar/UtamPageObject_Tests.java

```diff
@@ -175,4 +179,29 @@ public void testNotAllowedPropertyThrows() {
         containsString("error 200: root elements: incorrect format of elements: \n"
             + "Unrecognized field \"wrong\""));
   }
+
+  @Test
+  public void testMetadataProperty() {
+    String json = "{ \"metadata\": {}}";
+    assertThat(
+            new DeserializerUtilities().getResultFromString(json).getPageObject(),
+            is(not(nullValue())));
+  }
+
+  @Test
+  public void testMetadataPropertyValueNotObjectThrows() {
+    List<String> jsonValues = Arrays.asList(
+        "{ \"metadata\": \"value\" }",
+        "{ \"metadata\": [\"value\", \"anotherValue\"] }",
+        "{ \"metadata\": 0 }",
+        "{ \"metadata\": true }",
+        "{ \"metadata\": null }"
+    );
+    jsonValues.forEach((json) -> {
+      UtamError e = expectThrows(UtamError.class,
+          () -> new DeserializerUtilities().getResultFromString(json));
+      assertThat(e.getMessage(), containsString(
+          "metadata property value must be an object"));
```

Comment:
pls add err code to expected string, it's easier to track then what's tested

Reply:
Done!